include/pini/util.hpp

```diff
@@ -10,10 +10,11 @@
 
 namespace util {
 // return vector of strings containing each line of the given file
-std::vector<std::string> get_lines(std::filesystem::path const &filename);
+std::vector<std::string> get_lines(std::filesystem::path const& filename);
 
 // extract key-value pairs from given vector of strings and insert them in an
 // unordered_map
-std::unordered_map<std::string, std::string>
-insert_pairs(std::vector<std::string> file_lines);
-} // namespace util
\ No newline at end of file
+std::unordered_map<std::string, std::string> insert_pairs(std::vector<std::string> file_lines);
+
+std::string trim_whitespace(std::string const& str);
```

Comment:
Why not use string_views for all such operations and avoid pointless allocations etc in the middle of substringing?